### DISCUSSION TITLE
Cannot find Parcel plugin "@parcel/transformer-image"

### DISCUSSION THREAD

**Discussion by lgarron** [parcel-bundler/parcel#6163] [General] [answered]
# 🐛 bug report

Installing the latest nightly version Parcel breaks importing images.

## 🎛 Configuration (.babelrc, package.json, cli command)

https://github.com/cubing/cubing.js/blob/main/package.json

## 🤔 Expected Behavior

The project builds.

## 😯 Current Behavior

```
      /Users/lgarron/Code/git/github.com/cubing/cubing.js/node_modules/@parcel/config-default/index.json:38:35
  37 |     "custom:*.vue": ["@parcel/transformer-vue"],
> 38 |     "url:*.{png,jpg,jpeg,webp}": ["@parcel/transformer-image"],
>    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot find module "@parcel/transformer-image", did you mean "@parcel/transformer-babel"?
  39 |     "url:*": ["@parcel/transformer-raw"]
  40 |   },
```

<img width="1281" alt="Screen Shot 2021-04-22 at 01 53 55" src="https://user-images.githubusercontent.com/248078/115686002-9edb3300-a30d-11eb-9a18-d9cb811f4216.png">

## 💁 Possible Solution

It seems the image transformer is not installed in `node_modules` by default; perhaps it should be?

## 🔦 Context

I was trying to upgrade our Parcel version to see if potential fixes for other bugs are stable.

(Such as #3481. That one may be fixed, but while trying to create a minimal repro with Parcel 2.0.0-nightly.643... I ran into another infinite rebuild loop.)

## 💻 Code Sample

```
git clone https://github.com/cubing/cubing.js
cd cubing.js
git checkout a503e30ca3743742f2ef782898a51207e2c44504
npm install
npm install parcel@nightly
make dev
```

I'm sorry, I was unable to make a small repro.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.643+3baf990a |
| Node             | v15.12.0 | 
| npm/Yarn         | 7.6.3 |
| Operating System | macOS 11.2.3 |

**Comment by DeMoorJasper** [comment, marked as answer]
It's a very large dependency and a lot of people complain about install size so we've added something called `parcelDependencies` to config packages so it can be installed later and still match a certain version.

See https://github.com/parcel-bundler/parcel/blob/v2/packages/configs/default/package.json#L53

So you do need to install it manually, although it should install automatically in dev if I'm not mistaken.

**Reply by mischnic** [reply]
> automatically in dev 

It should always autoinstall, except for CI. (So both `parcel serve` and `parcel build`)

**Reply by lgarron** [reply]
Ah, that would explain it. We use `--no-autoinstall`.

**Reply by lgarron** [reply]
> Ah, that would explain it. We use `--no-autoinstall`.

It would be really nice if Parcel could warn that something is failing because of a required dependency that would be auto-installed.

I regularly work around Parcel issues and use `--no-autoinstall` to keep the build process more predictable (e.g. so that it will work the the same on anyone's computer and offline), and a descriptive error message would have been really useful here.

**Reply by lgarron** [reply]
(How do I mark this discussion as answered?)

**Reply by mischnic** [reply]
> It would be really nice if Parcel could warn that something is failing because of a required dependency that would be auto-installed.

-> https://github.com/parcel-bundler/parcel/issues/6165

> (How do I mark this discussion as answered?)

<img width="980" alt="Bildschirmfoto 2021-04-22 um 11 41 34" src="https://user-images.githubusercontent.com/4586894/115692954-be9b4700-a35f-11eb-865e-70660db56493.png">
